uniffi_bindgen/src/bindings/kotlin/templates/RecordTemplate.kt

```diff
@@ -18,21 +19,28 @@ data class {{ type_name }} (
     {% endif %}
     companion object
 }
+{%- else -%}
+object {{ type_name }} {}
```

Comment:
Ok; I see what you're going for. There's a subtlety around `==` and `===` which is worth exploring here.

Using `object` is a singleton, rather than what I think we want here.

Reply:
You're right, I implemented what was requested in the issue since it was the simplest approach, but instantiating a normal class here here makes more sense, and will cause less code changes if some fields are added to the record later.

This is done in https://github.com/mozilla/uniffi-rs/pull/1781/commits/14836b0b30ebc4c48b3edc3c14cdaf9393fcf74e